### PR TITLE
SSCS-7300: Tech debt – SYA – Migrate to the latest health library and…

### PR DESCRIPTION
**Pull request by srochani**
… helm chart

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: appConfigurations.js [hunk 1]
```diff
@@ -89,6 +91,14 @@ const configureViews = app => {
   ]);
 };
 
+const client = ioRedis.createClient(
+  config.redis.host,
+  { enableOfflineQueue: false }
+);
+client.on('error', error => {
+  logger.trace(`Health check failed on redis ${error}`, 'health_check_error');
```
Comment (danhamilton) [inline appConfigurations.js:99]: do you want this as a trace or an error?

Would also be cool to push a 'event' to app insights so we can see how often redis connectivity is dropping

Comment (srochani) [reply appConfigurations.js:99]: This trace is for appInsight, we are using appInsight logger and trace is method of it

File: infrastructure/main.tf [hunk 2]
```diff
@@ -17,6 +17,26 @@ data "azurerm_subnet" "core_infra_redis_subnet" {
   resource_group_name  = "core-infra-${var.env}"
 }
 
+module "submit-your-appeal-frontend" {
```
Comment (timja) [inline infrastructure/main.tf:20]: why is this here?

File: services/healthcheck.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    },
+    timeout: config.health.timeout,
+    deadline: config.health.deadline
+  };
+};
+
+const setup = app => {
+  healthcheck.addTo(app, {
+    checks: {
+      redis: healthcheck.raw(() => client.ping().then(_ => healthcheck.status(_ === 'PONG'))
```
Comment (timja) [inline services/healthcheck.js:37]: move redis to readiness? cc @adusumillipraveen

File: infrastructure/main.tf (hunk 2 above)
Comment (srochani) [reply infrastructure/main.tf:20]: sorry I think by mistake I've added this, you are right its not needed for AKS. Let me removed it

File: services/healthcheck.js (hunk 3 above)
Comment (srochani) [reply services/healthcheck.js:37]: Thanks @timja and @adusumillipraveen as per ticket says to include this redis in default health and readiness
https://tools.hmcts.net/jira/browse/SSCS-7300
I've included into both endpoints

File: charts/sscs-tribunals-frontend/values.yaml [hunk 4]
```diff
@@ -4,6 +4,7 @@ nodejs:
   aadIdentityName: sscs
   environment:
     NODE_ENV: "production"
+    NODE_PATH: "."
```
Comment (timja) [inline charts/sscs-tribunals-frontend/values.yaml:7]: is this needed?

File: config/custom-environment-variables.json [hunk 5]
```diff
@@ -1,4 +1,9 @@
 {
+  "appName": "PACKAGES_NAME",
+  "project": "PACKAGES_PROJECT",
+  "environment": "NODE_ENV",
+  "version": "PACKAGES_VERSION",
+  "deployment_env": "DEPLOYMENT_ENV",
```
Comment (timja) [inline config/custom-environment-variables.json:6]: these shouldn't be needed, they are legacy vars from tactical
```suggestion
```

File: infrastructure/variables.tf [hunk 6]
```diff
@@ -8,6 +8,10 @@ variable "location" {
 
 variable "env" {}
 
+variable "sya_hostname" {
+  default = "localhost"
+}
```
Comment (timja) [inline infrastructure/variables.tf:13]:
```suggestion
```

File: infrastructure/variables.tf [hunk 7]
```diff
@@ -17,6 +21,10 @@ variable "node_environment" {
   default = "sandbox"
 }
 
+variable "node_path" {
+  default = "."
+}
```
Comment (timja) [inline infrastructure/variables.tf:26]:
```suggestion
```

File: charts/sscs-tribunals-frontend/values.yaml (hunk 4 above)
Comment (srochani) [reply charts/sscs-tribunals-frontend/values.yaml]: Not really, I've removed it

File: infrastructure/variables.tf (hunk 6 above)
Comment (srochani) [reply infrastructure/variables.tf]: Ok, removed

File: infrastructure/variables.tf (hunk 7 above)
Comment (srochani) [reply infrastructure/variables.tf]: Ok, removed

File: config/custom-environment-variables.json (hunk 5 above)
Comment (srochani) [reply config/custom-environment-variables.json]: Ok, removed

**Comment by srochani** [comment]
Closing and reopening the PR
